### PR TITLE
Determine app readiness based on deployment spec.Replicas rather than status.Replicas

### PR DESCRIPTION
**Pull request by g-linville**
Before this change, we were checking for app readiness by ensuring that the app's Deployment had matching `status.Replicas` and `status.ReadyReplicas`. If for some reason the Deployment fails to create any replicas (such as a mutating webhook that fails, if Istiod is down for example), then both `status.Replicas` and `status.ReadyReplicas` are set to 0, and Acorn thinks the app is ready even though it is not running at all.

With this change, we look at the Deployment spec instead to see the desired number of replicas.

### Checklist
- [x] The title of this PR would make a good line in Acorn's Release Note's Changelog
- [ ] The title of this PR ends with a link to the main issue being address in parentheses, like: `This is a title (#1216)`. [Here's an example](https://github.com/acorn-io/acorn/pull/1199)
- [ ] All relevant issues are referenced in the PR description. *NOTE: don't use [GitHub keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) that auto-close issues*
- [x] Commits follow [contributing guidance](https://github.com/acorn-io/acorn/blob/main/CONTRIBUTING.md#commits)
- [x] Automated tests added to cover the changes. If tests couldn't be added, an explanation is provided in the Verification and Testing section
- [x] Changes to user-facing functionality, API, CLI, and upgrade impacts are clearly called out in PR description
- [ ] PR has at least two approvals before merging (or a reasonable exception, like it's just a docs change)

### CODE REVIEW

File: pkg/controller/appstatus/status.go [hunk 1]
```diff
@@ -481,7 +481,11 @@ func AppStatus(req router.Request, resp router.Response) error {
 
 		status := container[containerName]
 		status.Ready = dep.Status.ReadyReplicas
-		status.ReadyDesired = dep.Status.Replicas
+		if dep.Spec.Replicas != nil {
```
Comment (StrongMonkey) [inline pkg/controller/appstatus/status.go:484]: Hmm this should be dep.status.Replica?

Comment (g-linville) [reply pkg/controller/appstatus/status.go:484]: No, the `status.Replicas` has the possibility to be 0 (not set) if something goes wrong. `spec.Replicas` is a pointer so I have to do the nil check first before proceeding.

Comment (StrongMonkey) [reply pkg/controller/appstatus/status.go:484]: But dep.spec.replicas will never be the replica that is ready? It is the desired replica for a deployment, not the ready replica.

Comment (StrongMonkey) [reply pkg/controller/appstatus/status.go:484]: oh wait, this is `ReadyDesired`.... yeah setting spec.replica makes sense...

**Review by tylerslaton — APPROVED** [review]

File: pkg/controller/appstatus/status.go [hunk 2]
```diff
@@ -481,7 +481,11 @@ func AppStatus(req router.Request, resp router.Response) error {
...
+			status.ReadyDesired = *dep.Spec.Replicas
+		} else {
+			status.ReadyDesired = 0
```
[the remark this answers is not in the document]
Comment (g-linville) [reply pkg/controller/appstatus/status.go]: Good call, fixed :+1:

**Review by iwilltry42 — APPROVED** [review]

**Pull request merged**